src/dev/lupio/lupio_tmr.cc

```diff
@@ -49,26 +49,30 @@ LupioTMR::LupioTMR(const Params &params) :
     BasicPioDevice(params, params.pio_size),
     system(params.system),
     nThread(params.num_threads),
-    tmrEvent([this]{ lupioTMRCallback(); }, name()),
     intType(params.int_type)
 {
+    _timers.resize(nThread);
+
+    for (int cpu = 0; cpu < nThread; cpu++) {
+        _timers[cpu].tmrEvent = new EventFunctionWrapper(
```

Comment:
Where do you call delete? You should probably add a destructor for either the `LupioTimer` struct or for the `LupioTMR` which deletes this dynamically allocated memory.